### PR TITLE
Feat: start new front

### PR DESCRIPTION
**Pull request by ga-devfront**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Creating _dev folder to build scss and ts with vite. CI and storybook are updated too.
| Type?             | improvement
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | --
| Sponsor company   | @PrestaShopCorp
| How to test?      | Just follow the readme for module and/or storybook to test if all it's fine.

### CODE REVIEW

File: views/templates/layouts/welcome.html.twig [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{% block title %}
+  <h1>{{ 'Welcome to PrestaShop Update Assistant'|trans({}) }}</h1>
+{% endblock %}
+
+{% block content %}
+  <div class="welcome-page__card-list">
+    {% include '@ModuleAutoUpgrade/components/radio-card.html.twig' with {
+      checked: true,
+      title: "Update your store",
+      message: "Update your store to benefit from the latest improvements, bug fixes and security patches.",
```
Comment (Quetzacoalt91) [inline views/templates/layouts/welcome.html.twig:16]: Should be translatable

File: views/templates/layouts/welcome.html.twig [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      badgeLabel: "",
+      releaseNote: "",
+    } %}
+    {% include '@ModuleAutoUpgrade/components/radio-card.html.twig' with {
+      checked: true,
+      title: "Restore from a backup",
+      message: "Use this feature if the update failed or if your store or a  module is no longer working properly.",
```
Comment (Quetzacoalt91) [inline views/templates/layouts/welcome.html.twig:23]: Should be translatable

File: .github/workflows/build-release.yml [hunk 3]
```diff
@@ -20,10 +20,22 @@ jobs:
       - name: Install composer dependencies
         run: composer install --ansi --prefer-dist --no-interaction --no-progress --no-dev
 
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.11.0'
```
[the remark this answers is not in the document]
Comment (ga-devfront) [reply .github/workflows/build-release.yml:26]: switch to '20.x'

File: .github/workflows/js.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    name: JS linter syntax check
+    runs-on: unbuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.11.0'
```
[the remark this answers is not in the document]
Comment (ga-devfront) [reply .github/workflows/js.yml:17]: switch to '20.x'

File: views/templates/modals/modal-error-report.html.twig [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+			</p>
+		{% endif %}
+	</div>
+{% endblock %}
+
+{% block modal_extra_content_inner %}
+	<div class="modal__error-report-form">
+		<div class="form-group">
+			<label for="userEmail">Email (optional)</label>
+			<input type="password" class="form-control" id="userEmail" placeholder="Enter your email address">
```
[the remark this answers is not in the document]
Comment (ga-devfront) [reply views/templates/modals/modal-error-report.html.twig:22]: We going to add translation then we integrate the new pages.

File: views/templates/modals/modal-error-report.html.twig [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+				</a>
+			</p>
+		{% endif %}
+	</div>
+{% endblock %}
+
+{% block modal_extra_content_inner %}
+	<div class="modal__error-report-form">
+		<div class="form-group">
+			<label for="userEmail">Email (optional)</label>
```
[the remark this answers is not in the document]
Comment (ga-devfront) [reply views/templates/modals/modal-error-report.html.twig:21]: same

File: views/templates/modals/modal-error-report.html.twig [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+
+{% block modal_content %}
+	<div class="modal__error-report-content">
+		<p>{{ 'Help us improve the module by sending us this error report. You can add details in the description if you want to.'|trans({}) }}</p>
+
+		{% if dataPrivacyLink %}
+			<p>
+				<a class="link" href="{{ dataPrivacyLink }}">
+					Learn more about your data and your rights<i class="material-icons">launch</i>
```
[the remark this answers is not in the document]
Comment (ga-devfront) [reply views/templates/modals/modal-error-report.html.twig:11]: same

File: views/templates/layouts/welcome.html.twig (hunk 1 above)
Comment (ga-devfront) [reply views/templates/layouts/welcome.html.twig:16]: same

File: views/templates/layouts/welcome.html.twig (hunk 2 above)
Comment (ga-devfront) [reply views/templates/layouts/welcome.html.twig:23]: same

**Review by M0rgan01 — APPROVED** [review]

**Review by Quetzacoalt91 — APPROVED** [review]

File: controllers/admin/AdminSelfUpgradeController.php [hunk 8]
```diff
@@ -526,6 +528,35 @@ public function initContent()
...
+
         return parent::initContent();
     }
+
+    /**
+     * @return void
+     */
+    private function addNewUIAssets()
+    {
+        if (!empty($_ENV['AUTOUPGRADE_DEV_WATCH_MODE']) && $_ENV['AUTOUPGRADE_DEV_WATCH_MODE'] === '1') {
```
Comment (Hlavtox) [inline controllers/admin/AdminSelfUpgradeController.php:541]: Not sure if this should be in the module files guys, isn't there some other way to run the dev environment?

Comment (ga-devfront) [reply controllers/admin/AdminSelfUpgradeController.php:541]: Unfortunately the Vite dev mode (_dev folder) has no influence on the symfony part, that's why I decided to add an env file to define it manually. But this has no impact on production.

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @ga-devfront 

Thank you for your PR, I tested it and it seems to works as you can see : 

[recording(286).webm](https://github.com/user-attachments/assets/4e814b01-619a-471c-a582-3ce904c17323)

Tested from : 
1.7.8.9 to 8.1.7 
8.0.4 to 8.1.7 
8.1.5 to 8.1.7 
8.1.7 to 9.0.0
8.1.5 to 9.0.0
8.0.4 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**